packages/typir/src/kinds/class/class-kind.ts

```diff
@@ -0,0 +1,198 @@
+/******************************************************************************
+ * Copyright 2024 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { assertUnreachable } from 'langium';
```

Comment:
Thanks, the splits for class and function are especially helpful, but I also like the new overall structure.